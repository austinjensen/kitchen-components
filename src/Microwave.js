import React, { Component } from 'react'

class Microwave extends Component {
    render(){
        console.log('Microwave.render()')
        return (
            <div>
                here is microwave
            </div>
        )
    }
    constructor() {
        super()
        console.log('Microwave.constructor()')
        
      }
      componentWillMount() {
        console.log('Microwave.componentWillMount()')
      }
      
      componentDidMount() {
        console.log('Microwave.componentDidMount()')
      }
}

export default Microwave