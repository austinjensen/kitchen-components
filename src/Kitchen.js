import React, { Component } from 'react'
import Microwave from './Microwave';
import Fridge from './fridge';
import Oven from './oven';

class Kitchen extends Component {
    render(){
        console.log('Kitchen.render()')
        return (
            <div>
                Kitchen is here
                <Microwave />
                <Oven/>
                <Fridge/>
            </div>
        )
    }
    constructor() {
        super()
        console.log('Kitchen.constructor()')
        
      }
      componentWillMount() {
        console.log('Kitchen.componentWillMount()')
      }
      
      componentDidMount() {
        console.log('Kitchen.componentDidMount()')
      }
}

export default Kitchen