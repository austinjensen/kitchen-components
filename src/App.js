import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';
import Kitchen from './Kitchen'

class App extends Component {
  constructor() {
    super()
    console.log('app.constructor()')
    
  }
  componentWillMount() {
    console.log('app.componentWillMount()')
  }
  
  componentDidMount() {
    console.log('app.componentDidMount()')
  }
  render() {
    console.log('app.render()')
    return (
      <div className="App">
        <Kitchen />
      </div>
    );
  }

}
export default App;
