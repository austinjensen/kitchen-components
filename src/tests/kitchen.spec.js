import React from 'react';
import Kitchen from '../Kitchen'
import Renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import Microwave from '../Microwave'
import App from '../App'
import Fridge from '../fridge'
import Oven from '../oven'

it('renders correctly', () => {
    const tree = Renderer
        .create(<Kitchen />)
        .toJSON()
    expect(tree).toMatchSnapshot()

})


it('it shallow renders correctly', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<Kitchen />)
    const result = renderer.getRenderOutput()

    // console.log(result)

    expect(result.type).toBe('div')
    expect(result.props.children).toEqual([
        'Kitchen is here',
        <Microwave/>,
        <Oven />,
        <Fridge/>
    ])
})

it('shallow render app correctly', () => {
    const renderer = new ShallowRenderer()
    renderer.render(<App />)
    const result = renderer.getRenderOutput()

    // console.log(result)

    expect(result.type).toBe('div')
    expect(result.props.children).toEqual(
        <Kitchen />
    )

})
