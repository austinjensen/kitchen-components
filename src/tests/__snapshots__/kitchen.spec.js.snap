// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Kitchen is here
  <div>
    here is microwave
  </div>
  <div>
    here is oven
  </div>
  <div>
    here is fridge
  </div>
</div>
`;
